addons/point_of_sale/models/pos_order.py

```diff
@@ -194,7 +194,7 @@ def _get_pos_anglo_saxon_price_unit(self, product, partner_id, quantity):
             .mapped('picking_id.move_lines')\
             .filtered(lambda m: m.product_id.id == product.id)\
             .sorted(lambda x: x.date)
-        price_unit = product._compute_average_price(0, quantity, moves)
+        price_unit = product.with_context(force_company=self.company_id.id)._compute_average_price(0, quantity, moves)
```

Comment:
are you sure `self` is always a singleton record? If yes, this method needs `self.ensure_one()`. If not, there is a probleme with your `.id`

Reply:
No, it's probably not **BUT**, the method is called from an `AccountMoveLine` method, see there at line 46: https://github.com/odoo/odoo/blob/905567b6de3ec7e48e2347d042b00482c9ab9c34/addons/point_of_sale/models/account_move.py#L39-L47

As you can see in this method, there is only one `aml`. 
From this `aml` we get the `move` and from it the `pos_orders`.

Considering that, I think it is safe to assume that all the `pos_order_ids` coming from there share the same `company_id` and thus is accessing id.
Thus, I can't do an `.ensure_one()` on the pos_order because it will break. I could do it on `self.company_id` but it would be dumb as accessing `id` on `company_id` will trigger `ensure_one` itself.